script.js

```diff
@@ -1,4 +1,183 @@
+    var rNG = randomNumberGenerator();
+    playerTwoRolls.push(rNG); //Stores numbers
+    console.log(`Rolling P2 number. P2 number(s) pushed: ${playerTwoRolls}`);
+    counter += 1;
+  }
+  playerTwoNumber = playerTwoRolls.join(""); //Joins the two numbers, pushes into playerOneNumber[] array
+};
+
+//RNG function.
+var randomNumberGenerator = function () {
+  var randomNumber = Math.random() * 6;
+  console.log(`Initial number: ${randomNumber}.`);
+  var numberCleanedUp = Math.floor(randomNumber) + 1; //+1 to ensure that '0' isn't rolled
+  console.log(`RNG working. Number generated: ${numberCleanedUp}`);
+  return numberCleanedUp;
+};
+
+//P1 Changing dice order function
+var playerOneShifting = function (input) {
+  console.log(
+    `P1 original order: ${playerOneRolls}. Current mode: ${currentMode}. Game type: ${gameType}.`
+  );
+  //Player 1 keeps order
+  if (currentMode == 2 && input == "") {
+    var myOutputValue = `Player one has chosen to keep the order of ${playerOneNumber}.<br/><br/>Player two, please enter "Swap" or leave the field blank and click submit to proceed.`;
+    console.log(`P1 keeping ${playerOneNumber}.`);
+  }
+  //Player 1 switches order
+  if (currentMode == 2 && input == "Swap") {
+    swapArray = [playerOneRolls[0], playerOneRolls[1]] = [
+      playerOneRolls[1],
+      playerOneRolls[0],
+    ]; //Swaps specific elements in array by making a new array (swapArray[]) and putting the elements in the new order
+    playerOneNumber = swapArray.join("");
+
+    console.log(`P1 numbers changed to ${playerOneNumber}.`);
+    var myOutputValue = `Player one has swapped to make ${playerOneNumber}.<br/><br/>Player two, please enter "Swap" or leave the field blank and click submit to proceed.`;
+  }
+  currentMode = 3; //Mode change - 3
+  console.log(`- Current mode: ${currentMode}. Game type: ${gameType}.`);
+  return myOutputValue;
+};
+
+//P2 Changing dice order function
+var playerTwoShifting = function (input) {
+  console.log(
+    `P2 original order: ${playerTwoRolls}. Current mode: ${currentMode}. Game type: ${gameType}.`
+  );
+  //Player 2 keeps order
+  if (currentMode == 3 && input == "") {
+    var myOutputValue = `Player two has chosen to keep the order of ${playerTwoNumber}.<br/><br/>Now, click the submit button to determine the winner!`;
+    console.log(`P2 keeping ${playerTwoNumber}.`);
+  }
+  //Player 2 switches order
+  if (currentMode == 3 && input == "Swap") {
+    currentMode = 4;
+    swapArray = [playerTwoRolls[0], playerTwoRolls[1]] = [
+      playerTwoRolls[1],
+      playerTwoRolls[0],
+    ];
+    playerTwoNumber = swapArray.join("");
+    console.log(`P2 numbers changed to ${playerTwoNumber}.`);
+    var myOutputValue = `Player two has swapped to make ${playerTwoNumber}.<br/><br/>Now, click the submit button to determine the winner!`;
+  }
+  currentMode = 4; //Mode change - 4
+  console.log(`- Current mode: ${currentMode}. Game type: ${gameType}.`);
+  return myOutputValue;
+};
```

Comment:
Same thing here. If you have functions that are doing the same things. Try to make a more flexile function that can be used by both or more players